### DISCUSSION TITLE
Lets use Ruff! (please vote)

### DISCUSSION THREAD

**Discussion by shroominic** [AnswerDotAI/fasthtml#561] [Polls]
Using Ruff is essential for maintaining clean, efficient Python code. As a fast, comprehensive linter, Ruff detects style violations, unused imports, and potential bugs, enforcing code quality standards and best practices. This reduces technical debt, minimizes runtime errors, and ensures code consistency, making it easier to read, maintain, and collaborate on.

**Comment by pydanny** [comment]
I voted against. I'm a fan of `ruff` for non-Answer.ai and non-Fast.ai projects. However...

For this project, besides everything in the [fastcore style guide](https://docs.fast.ai/dev/style.html) as someone with RSI, I'm happy to have a more terse project. There's also Guido's writings on [A foolish consistency is the hobgoblin of little minds](https://peps.python.org/pep-0008/#a-foolish-consistency-is-the-hobgoblin-of-little-minds) - there are reasons why this project doesn't follow the conventions in Ruff's tooling.
